rest/src/main/java/org/n52/web/ctrl/ParameterRequestMappingAdapter.java

```diff
@@ -52,13 +63,29 @@
 })
 public abstract class ParameterRequestMappingAdapter<T extends ParameterOutput> extends ParameterController<T> {
 
+    @Autowired
+    private EntityCounter counter;
+
     @Override
     @RequestMapping(path = "")
-    public ModelAndView getCollection(HttpServletRequest request,
-                                      HttpServletResponse response,
+    public ModelAndView getCollection(HttpServletResponse response,
                                       @RequestHeader(value = Parameters.HttpHeader.ACCEPT_LANGUAGE) String locale,
                                       @RequestParam MultiValueMap<String, String> query) {
-        return super.getCollection(request, response, locale, addHrefBase(query));
+        query = addHrefBase(query);
+        IoParameters queryMap = QueryParameters.createFromQuery(query);
+
+        if (queryMap.containsParameter("limit") || queryMap.containsParameter("offset")){
+            try {
+                OffsetBasedPagination obp = new OffsetBasedPagination(queryMap.getOffset(), queryMap.getLimit());
+                queryMap = queryMap.removeAllOf("limit").removeAllOf("offset");
+                Paginated<T> paginated = new Paginated(obp, this.getElementCount(queryMap));
+                this.addPagingHeaders(this.getCollectionPath(queryMap.getHrefBase()), response, paginated);
+            } catch (DataAccessException ex) {
```

Comment:
As above .. `DataAccessException` shouldn't be handled in the Web controller layer as it's specific to the DAO SPI implementation